.github/workflows/update-translations.yml

```diff
@@ -0,0 +1,38 @@
+on: workflow_dispatch
+
+name: Update transifex translations
+jobs:
+  update_translations:
+    name: Update transifex translations
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install transifex client
+        run: |
+          mkdir -p /home/runner/.local/bin
+          cd /home/runner/.local/bin
+          curl -o- https://raw.githubusercontent.com/transifex/cli/master/install.sh | bash
+          sudo apt-get install -y gettext
+      - name: Run update-translations script
+        # From https://github.com/marketplace/actions/create-pr-action:
+        # > executes an arbitrary command and commits the changes to the new pull request
+        uses: technote-space/create-pr-action@v2
+        env:
+          TRANSIFEX_TOKEN: ${{ secrets.TRANSIFEX_TOKEN }}
+          UPDATE_TRANSLATIONS_SKIP_GIT: 1
+        with:
+          EXECUTE_COMMANDS: |
+            git submodule update --init --recursive > /dev/null 2>&1
+            pip install --upgrade pip > /dev/null 2>&1
+            # install requirements, excluding difficult to install ones
+            grep -Ev 'xmlsec|python3-saml' requirements/dev-requirements.txt > ../requirements.txt
+            pip install -r ../requirements.txt  > /dev/null 2>&1
+            # copy .transifexrc.example to ~/.transifexrc, edited to include token
+            grep -v '^token =' .transifexrc.example > ~/.transifexrc
+            echo "token = ${TRANSIFEX_TOKEN}" >> ~/.transifexrc
```

Comment:
It would make this easier to read (and the build output would be better) if these were split up into separate steps. Looking at the command I can imagine 2 steps:
* install dependencies
* configure transifex

Reply:
The difficulty here is how this `technote-space/create-pr-action@v2` action is set up. It's set up to pull the repo for you (somewhat like `actions/checkout@v2`) as part of its run, and then let you modify any files you like in the git repo; then it uses the resulting diff to create a PR.

I can try and see what happens when the repo is already checked out earlier. If that works then I can split out the setup, which I agree would be ideal. My hunch is that this action might not account for the possibility and may just fail on clone if the repo already exists, or that the two actions will check out the repo in two different directory locations. But maybe I'll get lucky and it's well designed and they work well together.

Reply:
Let me know by the way if there's a better way you know of to open a PR from a github actions workflow. This is just some marketplace action that I found for doing this, not necessarily a consensus best practice.